src/sinks/azure_monitor_logs/sink.rs

```diff
@@ -0,0 +1,214 @@
+/// 9-digit nanosecond precision timestamps.
+#[derive(Clone, Debug)]
+struct JsonEncoding {
+    time_generated_key: Option<OwnedValuePath>,
+    encoder: (Transformer, Encoder<Framer>),
+}
+
+impl JsonEncoding {
+    pub fn new(transformer: Transformer, time_generated_key: Option<OwnedValuePath>) -> Self {
+        Self {
+            time_generated_key,
+            encoder: (
+                transformer,
+                Encoder::<Framer>::new(
+                    CharacterDelimitedEncoder::new(b',').into(),
+                    JsonSerializerConfig::default().build().into(),
+                ),
+            ),
+        }
+    }
+}
+
+impl crate::sinks::util::encoding::Encoder<Vec<Event>> for JsonEncoding {
+    fn encode_input(
+        &self,
+        mut input: Vec<Event>,
+        writer: &mut dyn io::Write,
+    ) -> io::Result<(usize, GroupedCountByteSize)> {
+        for event in input.iter_mut() {
+            let log = event.as_mut_log();
+
+            // `.remove_timestamp()` will return the `timestamp` value regardless of location in Event or
+            // Metadata, the following `insert()` ensures it's encoded in the request.
+            let timestamp = if let Some(Value::Timestamp(ts)) = log.remove_timestamp() {
+                ts
+            } else {
+                chrono::Utc::now()
+            };
+
+            if let Some(timestamp_key) = &self.time_generated_key {
+                log.insert(
+                    (PathPrefix::Event, timestamp_key),
+                    serde_json::Value::String(
+                        timestamp.to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
+                    ),
+                );
+            }
+        }
+
+        self.encoder.encode_input(input, writer)
+    }
+}
+
+struct AzureMonitorLogsRequestBuilder {
+    encoding: JsonEncoding,
+}
+
+impl RequestBuilder<Vec<Event>> for AzureMonitorLogsRequestBuilder {
+    type Metadata = EventFinalizers;
+    type Events = Vec<Event>;
+    type Encoder = JsonEncoding;
+    type Payload = Bytes;
+    type Request = AzureMonitorLogsRequest;
+    type Error = std::io::Error;
+
+    fn compression(&self) -> Compression {
+        Compression::None
+    }
+
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoding
+    }
+
+    fn split_input(
+        &self,
+        mut events: Vec<Event>,
+    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
+        let finalizers = events.take_finalizers();
+        let builder = RequestMetadataBuilder::from_events(&events);
+        (finalizers, builder, events)
+    }
+
+    fn build_request(
+        &self,
+        finalizers: Self::Metadata,
+        request_metadata: RequestMetadata,
+        payload: EncodeResult<Self::Payload>,
+    ) -> Self::Request {
+        AzureMonitorLogsRequest {
+            body: payload.into_payload(),
+            finalizers,
+            metadata: request_metadata,
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Could consider putting this and the tests in config.rs  in a `tests.rs` module.